### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by terminalnode**
Note: public repository

Moving from the Github template to an actual implementation.. lol.

Test deploys have been done, and can continue to be done, on [this branch](https://github.com/Viametrics/ViaMatrix/pull/468).

This first version still requires commit checkout, aws login and ecr login actions to precede it. Using these is all copy-paste though and available in the readme.

I think we can potentially do the ECR login ourselves. The ECR login action is actually quite complex, but a lot of it has to do with "is it a public ECR repo or nah?" etc. We only have private ones, and this action is specifically targeting private ECR repos.

Changing the flow to use Docker's new buildx thingie could also be neat, I guess.

**Comments on prettier/eslint configs will be rewarded with a slap in the face, but commits fixing them are welcome**

### CODE REVIEW

**Review by sjobergjim — APPROVED** [review]
lgtm, iiwiw =)

File: src/action-input2.ts
Comment (GergKllai1) [inline src/action-input2.ts:1]: Isn't this an exact copy of action-input.ts?
I don't see this being referenced in other code, why do we have it?

File: src/main.ts [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+	return `${actionInput.tagPrefix}${timestamp}-${shortSha}`;
 }
 
-run()
+try {
+	run();
+} catch (error) {
+	if (error instanceof Error) {
+		core.setFailed(error.message);
+	}
```
Comment (GergKllai1) [inline src/main.ts:36]: this looks odd is this needed?

**Review by GergKllai1 — APPROVED** [review]
LGTM I'll let Jim decide on the typescript.
Remove the input2 if not needed gg no re

**Pull request merged**